homeassistant/components/homekit_controller/climate.py

```diff
@@ -169,14 +169,51 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TARGET_HEATER_COOLER_STATE,
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
+            CharacteristicsTypes.TEMPERATURE_TARGET,
             CharacteristicsTypes.SWING_MODE,
         ]
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         temp = kwargs.get(ATTR_TEMPERATURE)
         state = self.service.value(CharacteristicsTypes.TARGET_HEATER_COOLER_STATE)
-        if state == TargetHeaterCoolerStateValues.COOL:
+
+        if state == TargetHeaterCoolerStateValues.AUTOMATIC:
+            chars: dict[str, Any] = {}
+
+            mode = MODE_HOMEKIT_TO_HASS[state]
+
+            if kwargs.get(ATTR_HVAC_MODE, mode) != mode:
+                mode = kwargs[ATTR_HVAC_MODE]
+                chars[CharacteristicsTypes.TARGET_HEATER_COOLER_STATE] = TARGET_HEATER_COOLER_STATE_HASS_TO_HOMEKIT[
+                    mode
+                ]
+
+            heat_temp = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            cool_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+
+            if (
+                (
+                    ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+                    & self.supported_features
+                )
+                and heat_temp
+                and cool_temp
+            ):
+                if temp is None:
+                    temp = (cool_temp + heat_temp) / 2
+                chars.update(
+                    {
+                        CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD: heat_temp,
+                        CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD: cool_temp,
+                        CharacteristicsTypes.TEMPERATURE_TARGET: temp,
```

Comment:
Does your device have all 3 of these? `TEMPERATURE_TARGET` isn't in the spec for `heater_cooler`.